content/articles/How-to-build-a-location-based-app-using-Django-and-GeoDjango/index.md

```diff
@@ -0,0 +1,240 @@
+            except KeyError:
+                pass 
+```
+
+We loop through the objects of the element containing the location and tags shop, inside the loop we extracted the name and coordinates. Then we return a valid GEOSGeometry object that is corresponding to spatial data. The with statement automatically close the file. We will call the function inside the Migration class.
+
+```
+operations = [
+        migrations.RunPython(load_data)
+    ]
+```
+Here is a copy of the migration file
+
+
+    from django.db import migrations  
+    import json  
+    from django.contrib.gis.geos import fromstr  
+    from pathlib import Path  
+    DATA_FILENAME = 'hotels.json'  
+    def load_data(apps, schema_editor):  
+    Hotel = apps.get_model('hotels', 'Hotel')  
+    jsonfile = Path(__file__).parents[2] / DATA_FILENAME  
+ 
+    with open(str(jsonfile)) as datafile:  
+        objects = json.load(datafile)  
+        for obj in objects['elements']:  
+            try:  
+                objType = obj['type']  
+                if objType == 'node':  
+                    tags = obj['tags']  
+                    name = tags.get('name','no-name')  
+                    longitude = obj.get('lon', 0)  
+                    latitude = obj.get('lat', 0)  
+                    location = fromstr(f'POINT({longitude} {latitude})', srid=4326)  
+                    Hotel(name=name, location=location).save()  
+            except KeyError:  
+                pass  
+    class Migration(migrations.Migration):  
+    dependencies = [  
+        ('hotels', '0001_initial'),  
+    ]  
+  
+    operations = [  
+        migrations.RunPython(load_data)  
+    ]
+
+So let's finish the migration by running:
+ ```
+python manage.py migrate
+```
+from the terminal.
+
+You can log in to the admin dashboard to view all the imported hotels.
+
+Right now we will display all the hotels using Django views, we will use the generic ListView class function. Let's open the hotels/views.py file and type some code.
+```
+from django.views import generic
+from django.contrib.gis.geos import fromstr
+from django.contrib.gis.db.models.functions import Distance
+from .models import Hotel
+```
+We will create a variable that stores the user current coordinates.
+
+    longitude = -80.191788  
+    latitude = 25.761681  
+      
+    user_coordinate = Point(longitude, latitude, srid=4326)
+
+Next, we will query our database for hotels that are within the user coordinate.
+
+    class HotelListView(generic.ListView):  
+        model = Hotel  
+        context_object_name = 'shops'  
+      queryset = Hotel.objects.annotate(distance=Distance('location', user_coordinate)).order_by('distance')[0:6]  
+
+
+The queryset uses the .annotate() to calculate the distance between the user coordinate and the nearest hotel. Next, we will create the home page where the user will see the hotels that are within his region. We will create the index.html file in hotels/templates/hotels/home_list.html
```

Comment:
This will improve readability and syntax highlighting with backtics.
Complete so that I can finish the preliminary review.

Reply:
I have effected the corrections.